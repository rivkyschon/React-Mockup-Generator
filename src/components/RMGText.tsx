// src/components/RMGText.tsx

import React from 'react';

interface RMGTextProps {
  content: string;
}

const RMGText: React.FC<RMGTextProps> = ({content}) => {
  return (
    <p className="text-gray-500 text-lg md:text-l font-normal leading-relaxed text-center">
      {content}
    </p>
  );
};

export default RMGText;
