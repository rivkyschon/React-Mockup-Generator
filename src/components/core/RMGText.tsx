// src/components/RMGText.tsx

import React from 'react';

interface RMGTextProps {
  content: string;
  className?: string;
}

const RMGText: React.FC<RMGTextProps> = ({content, className}) => {
  return (
    <p
      className={
        className ||
        'text-gray-500 text-lg md:text-l font-normal leading-relaxed text-center'
      }>
      {content}
    </p>
  );
};

export default RMGText;
