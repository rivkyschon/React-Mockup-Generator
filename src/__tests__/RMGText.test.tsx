// src/components/__tests__/RMGText.test.tsx

import React from 'react';
import {render} from '@testing-library/react';
import RMGText from '../components/core/RMGText';

describe('RMGText', () => {
  it('renders correctly', () => {
    const {asFragment} = render(<RMGText content="Test Content" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct content', () => {
    const {getByText} = render(<RMGText content="Test Content" />);
    // expect(getByText('Test Content')).toBeInTheDocument();
  });
});
