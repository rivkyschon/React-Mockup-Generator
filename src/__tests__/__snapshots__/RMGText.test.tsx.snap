// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RMGText renders correctly 1`] = `
<DocumentFragment>
  <p
    class="text-gray-500 text-lg md:text-l font-normal leading-relaxed text-center"
  >
    Test Content
  </p>
</DocumentFragment>
`;
